### PR TITLE
[SYCLomatic] Support migration of ld and st instructions

### PR DESCRIPTION
**Pull request by yihwang-nv**

### CODE REVIEW

File: clang/lib/DPCT/Asm/AsmNodes.cpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
+//===---------------------------- InlineAsm.cpp -----------------*- C++ -*-===//
```
Comment (tangjj11) [inline clang/lib/DPCT/Asm/AsmNodes.cpp]:
```suggestion
//===---------------------------- AsmNodes.cpp
 -----------------*- C++ -*-===//
```

File: clang/lib/DPCT/Asm/AsmNodes.cpp [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+//
+//===----------------------------------------------------------------------===//
+
+#include "AsmNodes.h"
+
+using namespace clang::dpct;
+
+InlineAsmType::~InlineAsmType() = default;
+InlineAsmDecl::~InlineAsmDecl() = default;
+// PTXNamedDecl::~PTXNamedDecl() = default;
```
Comment (tangjj11) [inline clang/lib/DPCT/Asm/AsmNodes.cpp]: what is this line.

File: clang/lib/DPCT/Asm/AsmNodes.h [hunk 3]
```diff
@@ -0,0 +1,836 @@
+//===---------------------------- InlineAsm.h -------------------*- C++ -*-===//
```
Comment (tangjj11) [inline clang/lib/DPCT/Asm/AsmNodes.h]:
```suggestion
//===---------------------------- AsmNodes.h -------------------*- C++ -*-===//
```

File: clang/lib/DPCT/Asm/AsmNodes.h [hunk 4]
```diff
@@ -0,0 +1,836 @@
...
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef CLANG_DPCT_INLINE_ASM_H
```
Comment (tangjj11) [inline clang/lib/DPCT/Asm/AsmNodes.h]: use wrong macro name.

**Review by tangjj11 — CHANGES_REQUESTED** [review]

File: clang/lib/DPCT/Asm/AsmNodes.cpp (hunk 1 above)
Comment (yihwang-nv) [reply clang/lib/DPCT/Asm/AsmNodes.cpp]: done

File: clang/lib/DPCT/Asm/AsmNodes.cpp (hunk 2 above)
Comment (yihwang-nv) [reply clang/lib/DPCT/Asm/AsmNodes.cpp]: removed

File: clang/lib/DPCT/Asm/AsmNodes.h (hunk 3 above)
Comment (yihwang-nv) [reply clang/lib/DPCT/Asm/AsmNodes.h]: done

File: clang/lib/DPCT/Asm/AsmNodes.h (hunk 4 above)
Comment (yihwang-nv) [reply clang/lib/DPCT/Asm/AsmNodes.h]: done

**Review by tangjj11 — APPROVED** [review]

File: clang/test/dpct/asm/st.cu [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+// RUN: dpct --format-range=none -out-root %T/st %s --cuda-include-path="%cuda-path/include" -- -std=c++14 -x cuda --cuda-host-only
+// RUN: FileCheck %s --match-full-lines --input-file %T/st/st.dp.cpp
+// RUN: %if build_lit %{icpx -c -fsycl %T/st/st.dp.cpp -o %T/st/st.dp.o %}
+
+// clang-format off
+#include <cuda_runtime.h>
+
+__global__ void st(int *a) {
+  // CHECK: *a = 111;
+  asm volatile ("st.global.s32 [%0], %1;" :: "l"(a), "r"(111));
```
Comment (zhimingwang36) [inline clang/test/dpct/asm/st.cu:22]: could you list all kind of the test that not support yet as test without checking,  then it easy to understand what is not supported yet. (try to understand what the "partial" mean in inc file.)

Comment (yihwang-nv) [reply clang/test/dpct/asm/st.cu:22]: added

**Review by zhimingwang36 — APPROVED** [review]

**Pull request merged**